### ISSUE TITLE
make sure subsystems are tolerant to duplicate ViewUpdates

### ISSUE THREAD

**Issue by ordian** [paritytech/polkadot#2133]
after https://github.com/paritytech/polkadot/pull/2128, we might indeed have a situation like 
https://github.com/paritytech/polkadot/pull/1741#discussion_r493677428
i.e. a `ViewUpdate` event with the same heads when a block is finalized

it should be tolerated by subsystems and we shouldn't punish peers for the "same but different" view updates

**Comment by rphmeier** [comment]
triage?

**Comment by ordian** [comment]
I believe this is no longer relevant after #2233, i.e. we no longer send view updates that update only finalized_number.

**Comment by ordian** [comment]
However, technically we allow to receive such updates, maybe we should simply skip such updates here https://github.com/paritytech/polkadot/blob/b0ff3d49ed96293bc1055e3e268e5dd2df939d2f/node/network/bridge/src/lib.rs#L496

(by comparing [sorted] heads of the views instead of the views themselves)
